### PR TITLE
Implements 80 new MMX instructions

### PR DESCRIPTION
**Pull request by Sonicadvance1**
Pulls in #70 and #72 as requirements first
This is almost all of the MMX instructions
Main ones missing are some SSE<->MMX interop instructions, integer average, and move with mask selection.

### CODE REVIEW

**Comment by skmp** [comment]
Do the tests cover all of the instructions?

**Comment by Sonicadvance1** [comment]
> Do the tests cover all of the instructions?

Yes these unit tests cover all the MMX instructions that are supported.
Number of instructions to unit tests disparity comes down to comes down to a few instructions testing multiple opcodes going either direction.

File: External/FEXCore/Source/Interface/Core/Frontend.cpp [hunk 1]
```diff
@@ -71,10 +71,32 @@ static uint32_t MapModRMToReg(uint8_t REX, uint8_t bits, bool HighBits, bool Has
...
+  constexpr std::array<uint64_t, 16> MMIndexes = {
+    FEXCore::X86State::REG_MM_0,
+    FEXCore::X86State::REG_MM_1,
+    FEXCore::X86State::REG_MM_2,
+    FEXCore::X86State::REG_MM_3,
+    FEXCore::X86State::REG_MM_4,
+    FEXCore::X86State::REG_MM_5,
+    FEXCore::X86State::REG_MM_6,
+    FEXCore::X86State::REG_MM_7,
+    ~0ULL,
```
[the remark this answers is not in the document]
Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/Core/Frontend.cpp:83]: True, I'll fix that tomorrow.

File: External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp [hunk 2]
```diff
@@ -5031,6 +5054,129 @@ void OpDispatchBuilder::PMULLOp(OpcodeArgs) {
...
 
+template<bool ToXMM>
+void OpDispatchBuilder::MOVQ2DQ(OpcodeArgs) {
+  OrderedNode *Src = LoadSource(FPRClass, Op, Op->Src[0], Op->Flags, -1);
+
+  // This instruction is a bit special in that if the source is MMX then it zexts to 128bit
+  if (ToXMM) {
+    _StoreContext(FPRClass, 8, offsetof(FEXCore::Core::CPUState, xmm[Op->Dest.TypeGPR.GPR - FEXCore::X86State::REG_XMM_0][0]), Src);
+    auto Const = _Constant(0);
+    _StoreContext(GPRClass, 8, offsetof(FEXCore::Core::CPUState, xmm[Op->Dest.TypeGPR.GPR - FEXCore::X86State::REG_XMM_0][1]), Const);
```
[the remark this answers is not in the document]
Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp]: Fixed!

File: External/FEXCore/Source/Interface/Core/Frontend.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/Core/Frontend.cpp]: Fixed!

**Pull request merged**